### PR TITLE
CLI changes required for `to` account signing

### PR DESCRIPTION
**Pull request by ParthDesai**
#### Problem
See #6449 

#### Summary of Changes
CLI changes required to make #6449 to work

Fixes #

### CODE REVIEW

File: cli/src/stake.rs [hunk 1]
```diff
@@ -706,36 +709,62 @@ pub fn process_delegate_stake(
 mod tests {
     use super::*;
     use crate::cli::{app, parse_command};
+    use solana_sdk::signature::{gen_keypair_file, read_keypair_file};
+
+    fn make_tmp_path(name: &str) -> String {
```
Comment (mvines) [inline cli/src/stake.rs]: can we `use tempfile::TempDir;`?

Comment (ParthDesai) [reply cli/src/stake.rs]: @mvines done

File: cli/src/storage.rs [hunk 2]
```diff
@@ -226,50 +228,67 @@ pub fn process_show_storage_account(
...
                 command: CliCommand::CreateStorageAccount {
                     account_owner: pubkey,
-                    storage_account_pubkey,
+                    storage_account: storage_account_keypair,
                     account_type: StorageAccountType::Archiver,
                 },
                 require_keypair: true
             }
         );
+        tmp_file.close().unwrap();
```
Comment (mvines) [inline cli/src/storage.rs]: nit: Is this even needed?  When `tmp_file` is dropped it should be deleted

File: cli/src/storage.rs [hunk 3]
```diff
@@ -226,50 +228,67 @@ pub fn process_show_storage_account(
...
                 command: CliCommand::CreateStorageAccount {
                     account_owner: pubkey,
-                    storage_account_pubkey,
+                    storage_account: storage_account_keypair,
                     account_type: StorageAccountType::Validator,
                 },
                 require_keypair: true
             }
         );
+        tmp_file.close().unwrap();
```
Comment (mvines) [inline cli/src/storage.rs]: nit: same, maybe not needed

**Comment by ParthDesai** [comment]
As per tmpfile doc, close is need to catch any error we encountered while
deleting the file, while destructor will obviously siltenly ignore those
errors.

So, wanted to err on the side of caution.

**Comment by mvines** [comment]
Not sure what we'd do about an error when a file fails to be deleted but ok sure.  If/when that happens we'd probably "fix" by removing the explicit close

**Comment by ParthDesai** [comment]
> Not sure what we'd do about an error when a file fails to be deleted but ok sure. If/when that happens we'd probably "fix" by removing the explicit close

After some thinking, I came to conclusion that you are right. Not able to delete the temp file is system level problem, for which we cannot fail our tests.

**Pull request merged**